syft/file/digest_cataloger.go

```diff
@@ -22,13 +29,22 @@ func NewDigestsCataloger(hashes []crypto.Hash) (*DigestsCataloger, error) {
 
 func (i *DigestsCataloger) Catalog(resolver source.FileResolver) (map[source.Location][]Digest, error) {
 	results := make(map[source.Location][]Digest)
+	var locations []source.Location
 	for location := range resolver.AllLocations() {
+		locations = append(locations, location)
+	}
+	stage, prog := digestsCatalogingProgress(int64(len(locations)))
+	for location := range resolver.AllLocations() {
```

Comment:
It looks like we're ranging through `resolver.AllLocations()` twice now. At first it wasn't obvious why, but I'm now thinking it's because we can't get a slice upfront from the `resolver` object, and then we need to access the total count of locations for the sake of reporting to the UI. Is that right?

Reply:
that's exactly right; I considered changing the resolver to return the slice, but decided against making an interface change for a UI enhancement.

Reply:
Makes sense. A "nit" comment would be to range through the `locations` instead (in the second loop) since that collection is already available in scope, rather than access the collection via multiple ways in the same set of lines, but that's an optional comment wrt this PR 👍

Reply:
good call! That was an oversight, I'll adjust